src/clarity-angular/layout/vertical-nav/vertical-nav.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Component, ElementRef, EventEmitter, HostListener, Input, OnDestroy, Output} from "@angular/core";
+import {VerticalNavService} from "./providers/vertical-nav.service";
+import {Subscription} from "rxjs/Subscription";
+
+@Component({
+    selector: "clr-vertical-nav",
+    templateUrl: "./vertical-nav.html",
+    providers: [VerticalNavService],
+    host: {
+        "class": "clr-vertical-nav",
+        "[class.is-collapsed]": "collapsed",
+        "[class.has-nav-groups]": "hasNavGroups",
+        "[class.has-icons]": "hasIcons"
+    }
+})
+export class VerticalNav implements OnDestroy {
+
+    get collapsible(): boolean {
+        return this._verticalNavService.collapsible;
+    }
+
+    @Input("clrVerticalNavCollapsible")
+    set collapsible(value: boolean) {
+        this._verticalNavService.collapsible = value;
+    }
+
+    get collapsed(): boolean {
+        return this._verticalNavService.collapsed;
+    }
+
+    @Input("clrVerticalNavCollapsed")
+    set collapsed(value: boolean) {
+        this._verticalNavService.collapsed = value;
+    }
+
+    @Output("clrVerticalNavCollapsedChange")
+    private _collapsedChanged: EventEmitter<boolean> = new EventEmitter<boolean>(true);
+
+    get hasNavGroups(): boolean {
+        return this._verticalNavService.navGroups.length > 0;
+    }
+
+    get hasIcons(): boolean {
+        return this._verticalNavService.hasIcons;
+    }
+
+    get angleDoubleDirection(): string {
+        return (this.collapsed) ? "right" : "left";
+    }
+
+    private _sub: Subscription;
+
+    constructor(private _verticalNavService: VerticalNavService, private elRef: ElementRef) {
+        this._sub = this._verticalNavService.collapsedChanged.subscribe(value => {
+            this._collapsedChanged.emit(value);
+        });
+    }
+
+    toggleByButton() {
+        this.collapsed = !this.collapsed;
+    }
+
+    @HostListener("click", ["$event.target"])
+    onHostClick(target: any): void {
+        if (this.collapsed && (target === this.elRef.nativeElement)) {
+            this.collapsed = false;
+        }
```

Comment:
You are planning on adding an extra container around the links. I just realized that it will break this.